### ISSUE TITLE
fix: bitcoind version breaks some recent psbt tests

### ISSUE THREAD

**Issue by m-schmoock** [ElementsProject/lightning#4144]
### Issue and Steps to Reproduce
Running the recently added PSBT tests on a pre 0.20 `bitcoind` seems to fail. Whereas they work with recent versions of `bitcoind` (At least they work with `0.20.1` but not `0.19.1`)

Affected tests:
```
tests/test_wallet.py::test_fundpsbt
tests/test_wallet.py::test_utxopsbt
tests/test_wallet.py::test_sign_and_send_psbt
```

### Message:
```
bitcoin.rpc.JSONRPCError: {'code': -22, 'message': 'TX decode failed PSBT is not sane.: iostream error'}
```

**Comment by darosior** [comment]
Must be due to some fixes in https://bitcoincore.org/en/releases/0.20.0/ or https://bitcoincore.org/en/releases/0.20.1/ to RPC calls used in these tests. AFAICT this does not affect our minimum supported bitcoind version.

**Comment by niftynei** [comment]
This is working as intended. See https://github.com/bitcoin/bitcoin/pull/19215

**Comment by niftynei** [comment]
tl;dr: we add both witness-utxo and non-witness-utxo data to PSBTs, which any bitcoind prior to 0.20.1\* fails as invalid due to the strictness of the parser contained therein.

\*  might be v0.20.0, not sure when that PR got merged in bitcoind

**Comment by niftynei** [comment]
fwiw this is currently a problem for the elementsd integration which is still on v0.18.0. I've notified the maintainers and it should be fixed/updated when they update elementsd to support the latest.

**Comment by m-schmoock** [comment]
@niftynei we could disable the test on older bitcoind versions as well...

**Comment by niftynei** [comment]
No, older versions of bitcoind are incompatible.  if you’re running an
older version of bitcoind, you need to upgrade to use newer versions of
c-lightning.

Skipping the test merely hides this fact, and makes it likely you’ll have
the same problem elsewhere later.

**Comment by m-schmoock** [comment]
Okay, then we should update the Readme ;) didn't know that

**Comment by m-schmoock** [comment]
@niftynei also, we should stop the daemon from running on older versions if thats true.

**Comment by darosior** [comment]
I think niftynei meant for development.

Users can still run fine with 0.16 mini, we don't make PSBT calls to bitcoind in this case :)

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by m-schmoock** [comment]
Isn't that PSBT stuff for dual funding and such? Thats currently experimental.. so yes, maybe its currently only developers. but that will soon change then I guess..

**Comment by darosior** [comment]
> Isn't that PSBT stuff for dual funding and such?

iirc it's not required by the specs but eases management

> but that will soon change then I guess..

I don't expect us to use more calls to bitcoin-cli, we finalize the PSBT before broadcasting anyways

**Comment by m-schmoock** [comment]
Lets give @niftynei the final call to close this issue then :D
